src/Validator/Rules/UniqueEnumValueNames.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GraphQL\Validator\Rules;
+
+use GraphQL\Error\Error;
+use GraphQL\Language\AST\EnumTypeDefinitionNode;
+use GraphQL\Language\AST\EnumTypeExtensionNode;
+use GraphQL\Language\AST\EnumValueNode;
+use GraphQL\Language\AST\Node;
+use GraphQL\Language\AST\NodeKind;
+use GraphQL\Language\Visitor;
+use GraphQL\Language\VisitorOperation;
+use GraphQL\Type\Definition\EnumType;
+use GraphQL\Validator\SDLValidationContext;
+
+class UniqueEnumValueNames extends ValidationRule
+{
+    public function getSDLVisitor(SDLValidationContext $context): array
+    {
+        /** @var array<string, array<string, EnumValueNode>> $knownValueNames */
+        $knownValueNames = [];
+
+        /**
+         * @param (Node&EnumTypeDefinitionNode)|(Node&EnumTypeExtensionNode) $enum
+         */
+        $checkValueUniqueness = static function ($enum) use ($context, &$knownValueNames): VisitorOperation {
```

Comment:
```suggestion
        $checkValueUniqueness = static function (Node $enum) use ($context, &$knownValueNames): VisitorOperation {
```
Native type here?

Reply:
See https://github.com/webonyx/graphql-php/pull/995#discussion_r741482567

Reply:
I would feel safer with a native type, but am fine with relying on tests proving the correctness of this piece of code. I would rather not worsen performance.